src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -328,12 +335,45 @@ default boolean doesSneakBypassUse(ItemStack stack, net.minecraft.world.level.Le
     }
 
     /**
-     * Called to tick armor in the armor slot. Override to do something
+     * @deprecated Use {@link #inventoryTick(ItemStack, Level, ContextKey.Context)
      */
+    @Deprecated(forRemoval = true, since = "1.20.1")
     default void onArmorTick(ItemStack stack, Level level, Player player)
     {
     }
 
+    /**
+     * Called when this item is ticked in an inventory.<br>
+     * This is a more contextual version of {@link Item#inventoryTick(ItemStack, Level, Entity, int, boolean)} which allows providing arbitrary inventory-based context.
+     * <p>
+     * What context is provided is up to the caller, and is managed by the {@linkplain ContextKey context key system}.
+     * <p>
+     * In vanilla, the only inventory that ticks items is the player inventory.
+     * @param stack The stack being ticked.
+     * @param level The level of the of the ticking inventory. May be a client level.
+     * @param context The arbitrary context object. See {@link ContextKey}.
+     * @apiNote Override-Only, call via {@link IForgeItemStack#inventoryTick}
+     * 
+     * @see {@link InventoryTickContext} for the default context object.
+     */
+    @ApiStatus.OverrideOnly
+    @SuppressWarnings({"deprecation", "removal"})
+    default void inventoryTick(ItemStack stack, Level level, ContextKey.Context<?, ?> context)
+    {
+        if(context.ctx() instanceof VanillaCompatible vCtx) {
+            stack.inventoryTick(level, vCtx.getEntity(), vCtx.getSlot(), vCtx.isSelected());
+        }
+
+        // Vanilla always passes the "local" slotIndex, despite it being useless without the compartment.
+        context.ifPresent(InventoryTickContext.KEY, ctx -> {
```

Comment:
Should this be inside the previous if statement? I can see an argument to be made for either location.